### PR TITLE
DOC add priorities update 09/2024

### PR DESCRIPTION
**Pull request by glemaitre**
This the blog post to provide Probabl's OSS team update. It provides:

- [x] retrospective for the different packages maintained at Probabl
- [x] provide an update or recall prior objectives
- [x] provide an update on the focuses for each project

### CODE REVIEW

File: 09_2024_scikit_learn_priorities.md
Comment (adrinjalali) [inline 09_2024_scikit_learn_priorities.md]:
```suggestion
   for almost all estimators. It allows for fewer bugs and enables new
```

Comment (adrinjalali) [reply 09_2024_scikit_learn_priorities.md]: or everything except AdaBoost I think

File: 09_2024_scikit_learn_priorities.md
Comment (adrinjalali) [inline 09_2024_scikit_learn_priorities.md]: I feel like this could be restructured to avoid repeating the first sentence here as this:

```
- Short-Term priorities
  - ...
  - ...
- Mid-Term priorities
  - ...
  - ...
 ```

File: 09_2024_scikit_learn_priorities.md
Comment (adrinjalali) [inline 09_2024_scikit_learn_priorities.md:108]: Could also mention that we also pushed out a release with many compatibility fixes with dependency libraries.

File: 09_2024_scikit_learn_priorities.md
Comment (adrinjalali) [inline 09_2024_scikit_learn_priorities.md]:
```suggestion
## Our contrubotors
```
since not everyone is a maintainer?

File: 09_2024_scikit_learn_priorities.md
Comment (adrinjalali) [inline 09_2024_scikit_learn_priorities.md]: not sure in which section it goes, but we also have done quite a bit with the developer API work.

Comment (glemaitre) [reply 09_2024_scikit_learn_priorities.md]: Indeed. We did quite some work in between the moment I did the retrospective and the updated the new priority.

Comment (glemaitre) [reply 09_2024_scikit_learn_priorities.md]: Actually this is the last point raised in the previous item.

**Review by adrinjalali — APPROVED** [review]

File: 09_2024_scikit_learn_priorities.md
Comment (lorentzenchr) [inline 09_2024_scikit_learn_priorities.md]: This gives the impression that the list below was accomplished by probabl alone.

File: 09_2024_scikit_learn_priorities.md
Comment (lorentzenchr) [inline 09_2024_scikit_learn_priorities.md]: I would either expect "task is completed" or X out of Y remaining estimators where completed in the last 6 months.

File: 09_2024_scikit_learn_priorities.md
Comment (lorentzenchr) [inline 09_2024_scikit_learn_priorities.md:57]: Is it an estimator or a meta-estimator?
Should `FixedThresholdClassifier` be mentioned, too?

File: 09_2024_scikit_learn_priorities.md
Comment (lorentzenchr) [inline 09_2024_scikit_learn_priorities.md]: For example ...

File: 09_2024_scikit_learn_priorities.md
Comment (lorentzenchr) [inline 09_2024_scikit_learn_priorities.md:73]: Again, do you have numbers?

File: 09_2024_scikit_learn_priorities.md
Comment (lorentzenchr) [inline 09_2024_scikit_learn_priorities.md:77]: Was it together with Nathan from Quansight?

File: 09_2024_scikit_learn_priorities.md
Comment (lorentzenchr) [inline 09_2024_scikit_learn_priorities.md]: This sounds as if it was released the first time.

File: 09_2024_scikit_learn_priorities.md
Comment (lorentzenchr) [inline 09_2024_scikit_learn_priorities.md:157]: This is not statistical correctness, rather algorithmic/numerical.

File: 09_2024_scikit_learn_priorities.md
Comment (lorentzenchr) [inline 09_2024_scikit_learn_priorities.md:161]: More for my interest: Which ones?

File: 09_2024_scikit_learn_priorities.md
Comment (lorentzenchr) [inline 09_2024_scikit_learn_priorities.md]: I would mention the SLEP.

File: 09_2024_scikit_learn_priorities.md
Comment (lorentzenchr) [inline 09_2024_scikit_learn_priorities.md]: FYI: My benchmarking indicates that it is mostly the histogram computation.

File: 09_2024_scikit_learn_priorities.md
Comment (lorentzenchr) [inline 09_2024_scikit_learn_priorities.md]: Isn't this a community decision?

File: 09_2024_scikit_learn_priorities.md
Comment (lorentzenchr) [inline 09_2024_scikit_learn_priorities.md:191]: As I did in one of Adrin's PRs, it might be a good idea to contact relecant projects and involve them in the decisions.

**Review by lorentzenchr — COMMENTED** [review]
@glemaitre Thanks for sharing and giving the opportunity for feedback.

File: 09_2024_scikit_learn_priorities.md
Comment (glemaitre) [reply 09_2024_scikit_learn_priorities.md]: Of course. `Assess` here means that someone will report the pros/cons, potential difficulties, by looking in details if we want to go this road. The final call will be based on the consensus.

I will rephrase it because the `we should merge` does not reflect the consensus base decision.

File: 09_2024_scikit_learn_priorities.md
Comment (glemaitre) [reply 09_2024_scikit_learn_priorities.md]: I should decouple this point to have computation performance and statistical performance on the other side. We plan to look at both: using your benchmark and look at adding the second level parallelism as you mentioned. On the statistical side, Gael reported some benchmark where it seems that `HistGradientBoosting` is doing worse after tuning but this is not clear yet what is the reason. So we would like to have a factual answer to the reason there.

File: 09_2024_scikit_learn_priorities.md
Comment (glemaitre) [inline 09_2024_scikit_learn_priorities.md]:
```suggestion
for BLAS libraries (FlexiBLAS, OpenBLAS, Netlib, Accelerate, etc.).
```

File: 09_2024_scikit_learn_priorities.md
Comment (glemaitre) [reply 09_2024_scikit_learn_priorities.md]: What do you mean exactly? The support for FlexiBLAS has only been added lately.

File: 09_2024_scikit_learn_priorities.md
Comment (glemaitre) [reply 09_2024_scikit_learn_priorities.md]: I end up with a section to exactly addressed such impression. So I should move right at the beginning such that it is clear that this is not the case.

File: 09_2024_scikit_learn_priorities.md
Comment (lorentzenchr) [reply 09_2024_scikit_learn_priorities.md]: What you mean to say is "A new version of theadpoolctl has been released".

File: 09_2024_scikit_learn_priorities.md
Comment (glemaitre) [reply 09_2024_scikit_learn_priorities.md:73]: I'm not sure if we should specify numbers. Part of the info would be available here (https://blog.scikit-learn.org/resources/). The problem is that I don't know how to split external contributions from contribution from the core team.

Since we have the round-robin cycles, we almost ensure that all issues are triaged in this respect and it increases the hit rate on reviewing PRs.

Would you better like this explanation there or do you think of anyway to report proper numbers.

File: 09_2024_scikit_learn_priorities.md
Comment (glemaitre) [reply 09_2024_scikit_learn_priorities.md:77]: Indeed, it started with this [issue](https://github.com/scikit-learn/scikit-learn/issues/28978) and now Nathan is also involved in this [thread](https://github.com/scikit-learn/scikit-learn/issues/30007) regarding the stress-testing phase. I think Loïc is the person the most involved on this topic from our side.

Do you want me to have an explicit acknowledgement to the involvement of Nathan on this topic?

File: 09_2024_scikit_learn_priorities.md
Comment (glemaitre) [reply 09_2024_scikit_learn_priorities.md:161]: I'll ask @ogrisel to jump in because he would have a clearer idea than me :).

File: 09_2024_scikit_learn_priorities.md
Comment (lucyleeow) [inline 09_2024_scikit_learn_priorities.md]: Nitpick, "the operational decision" just sounds a bit odd (not sure why exactly, maybe the 'the' makes it sound like there is a specific decision you are referring to?), does "for operational decision-making" work?

File: 09_2024_scikit_learn_priorities.md
Comment (lucyleeow) [inline 09_2024_scikit_learn_priorities.md]:
```suggestion
      available in `scikit-learn` and making them available for third-party projects. For
```

File: 09_2024_scikit_learn_priorities.md
Comment (lucyleeow) [inline 09_2024_scikit_learn_priorities.md]:
```suggestion
      create estimators and check their compatibility with `scikit-learn`.
```

File: 09_2024_scikit_learn_priorities.md
Comment (lucyleeow) [inline 09_2024_scikit_learn_priorities.md]:
```suggestion
      started to improve the tests checking the API conformance of third-party libraries
```

File: 09_2024_scikit_learn_priorities.md
Comment (lucyleeow) [inline 09_2024_scikit_learn_priorities.md]: Optional suggestion
```suggestion
      to promote the latest work on `scikit-learn`;
      PyCon Lithuania, PyCon Italia, CZI Open Science, EuroSciPy, PyData
      Amsterdam, and PyData Paris.
```

File: 09_2024_scikit_learn_priorities.md
Comment (lucyleeow) [inline 09_2024_scikit_learn_priorities.md]:
```suggestion
orchestration: `joblib`, `loky`, `cloudpickle`, `threadpoolctl`, and worked on supporting
Python free-threaded mode.
```

File: 09_2024_scikit_learn_priorities.md
Comment (lucyleeow) [inline 09_2024_scikit_learn_priorities.md]:
```suggestion
We also helped with the release of `fairlearn` and made sure that it is compatible with the
```

File: 09_2024_scikit_learn_priorities.md
Comment (lucyleeow) [inline 09_2024_scikit_learn_priorities.md]:
```suggestion
The main activity in this project relates to non-trivial maintenance tasks
```

File: 09_2024_scikit_learn_priorities.md
Comment (lucyleeow) [inline 09_2024_scikit_learn_priorities.md]:
```suggestion
We made sure that `scikit-learn` is compatible with the WebAssembly stack:
```

File: 09_2024_scikit_learn_priorities.md
Comment (lucyleeow) [inline 09_2024_scikit_learn_priorities.md]:
```suggestion
     This will feed into the discussion to make a decision on the best way to move forward.
```

File: 09_2024_scikit_learn_priorities.md
Comment (lucyleeow) [inline 09_2024_scikit_learn_priorities.md]:
```suggestion
The main activity in this project relates to non-trivial maintenance tasks
```

File: 09_2024_scikit_learn_priorities.md
Comment (lucyleeow) [inline 09_2024_scikit_learn_priorities.md]:
```suggestion
We have to adapt the whole stack (`joblib`, `loky`, `cloudpickle`, `threadpoolctl`) to the
```

**Review by lucyleeow — COMMENTED** [review]
This was a very interesting read. Grammar nitpicks only, feel free to ignore anything that does not make sense.

**Comment by glemaitre** [comment]
Thanks @lucyleeow. I incorporated the feedback.

File: 09_2024_scikit_learn_priorities.md
Comment (glemaitre) [inline 09_2024_scikit_learn_priorities.md]:
```suggestion
      to use a pre-specified threshold for the operational decision-making.
```

File: 09_2024_scikit_learn_priorities.md
Comment (glemaitre) [reply 09_2024_scikit_learn_priorities.md]: I moved the acknowledgement right at the start.

File: 09_2024_scikit_learn_priorities.md
Comment (glemaitre) [reply 09_2024_scikit_learn_priorities.md]: I gave more straightforward explanation (only missing `AdaBoost`).

File: 09_2024_scikit_learn_priorities.md
Comment (glemaitre) [reply 09_2024_scikit_learn_priorities.md]: Done

File: 09_2024_scikit_learn_priorities.md
Comment (glemaitre) [reply 09_2024_scikit_learn_priorities.md:73]: I added more details regarding what we meant here.

File: 09_2024_scikit_learn_priorities.md
Comment (glemaitre) [reply 09_2024_scikit_learn_priorities.md:157]: Done

File: 09_2024_scikit_learn_priorities.md
Comment (glemaitre) [reply 09_2024_scikit_learn_priorities.md]: Done

File: 09_2024_scikit_learn_priorities.md
Comment (glemaitre) [reply 09_2024_scikit_learn_priorities.md]: Done

File: 09_2024_scikit_learn_priorities.md
Comment (glemaitre) [reply 09_2024_scikit_learn_priorities.md:191]: I'll mention this as well.

**Comment by glemaitre** [comment]
Going forward to merge this PR and start to work on the blog post. We can always amend if I get further comment :)

**Pull request merged**
